src/components/transport_manager/test/transport_adapter_test.cc

```diff
@@ -739,6 +740,43 @@ TEST_F(TransportAdapterTest, FindEstablishedConnection) {
   EXPECT_CALL(*serverMock, Terminate());
 }
 
+TEST_F(TransportAdapterTest, RunAppOnDevice_NoDeviseWithAskedId_UNSUCCESS) {
+  const std::string bundle_id = "test_bundle_id";
+
+  MockTransportAdapterImpl transport_adapter(
+      NULL, NULL, NULL, last_state_, transport_manager_settings);
+
+  utils::SharedPtr<MockDevice> mock_device =
+      utils::MakeShared<MockDevice>("test_device_name", "test_device_uid0");
+
+  ON_CALL(*mock_device, IsSameAs(_)).WillByDefault(Return(false));
+
+  transport_adapter.AddDevice(mock_device);
+
+  EXPECT_CALL(*mock_device, LaunchApp(bundle_id)).Times(0);
+
+  transport_adapter.RunAppOnDevice("test_device_uid1", bundle_id);
+}
+
+TEST_F(TransportAdapterTest, RunAppOnDevice_DeviseWithAskedIdWasFound_SUCCESS) {
+  const std::string bundle_id = "test_bundle_id";
+  const std::string device_uid = "test_device_uid";
+
+  MockTransportAdapterImpl transport_adapter(
+      NULL, NULL, NULL, last_state_, transport_manager_settings);
+
+  utils::SharedPtr<MockDevice> mock_device =
+      utils::MakeShared<MockDevice>("test_device_name", device_uid);
+
+  ON_CALL(*mock_device, IsSameAs(_)).WillByDefault(Return(false));
```

Comment:
@LuxoftAKutsan, @wolfylambova22, actually, I should remove this line. It's redundant, because on uninteresting call Google Mock will return false, and because in this test case, actually, there is nothing to compare with `mock_device`  in any case.